packages/ui/src/components/molecules/Dropdown/ListItemDropdown.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { useEffect } from 'react';
+import { DefaultTheme, styled } from 'styled-components';
+
+import { LangDisplay, RadioButton, Typography } from '../../atoms';
+
+interface ListItemDropdownProps {
+  text: string;
+  checkType?: string;
+  checked: boolean;
+  onChange: () => void;
+  focused?: boolean;
+}
+
+const LocalTypography = styled(Typography)`
+  color: ${({ theme }) => theme.palette.text.muted};
+`;
+
+interface ItemsProps {
+  checked: boolean;
+  $isFocused?: boolean;
+}
+
+const getBackgroundColor = (
+  theme: DefaultTheme,
+  checked: boolean,
+  $isFocused?: boolean,
+) => {
+  if ($isFocused || checked) {
+    return theme.palette.background.dropdownHover;
+  }
+  return theme.palette.background.separatorSecondary;
+};
+
+const Items = styled.div<ItemsProps>`
+  display: flex;
+  padding: 12px 24px;
+  align-items: center;
+  gap: 24px;
+  border-bottom: 1px solid ${({ theme }) => theme.palette.background.separator};
+  width: 100%;
+  cursor: pointer;
+  background: ${({ theme, checked, $isFocused }) =>
+    getBackgroundColor(theme, checked, $isFocused)};
+  ${LocalTypography} {
+    color: ${({ $isFocused, theme }) =>
+      $isFocused ? theme.palette.text.white : theme.palette.text.muted};
+  }
+  &:hover {
+    background-color: ${({ theme }) => theme.palette.background.dropdownHover};
+    ${LocalTypography} {
+      color: ${({ theme }) => theme.palette.text.white};
+    }
+  }
+  outline: none;
+
+  &:focus {
+    outline: none;
+  }
+`;
+
+export const ListItemDropdown: React.FC<ListItemDropdownProps> = ({
+  text: itemText,
+  checkType,
+  checked,
+  onChange,
+  focused = false,
+}) => {
+  const handleBoxClick = () => {
+    if (checkType && checkType === 'radio') {
+      onChange();
+    }
+  };
+
+  useEffect(() => {
+    if (focused) {
+      const divElement = document.getElementById(`item-${itemText}`);
+      divElement?.focus();
+    }
+  }, [focused, itemText]);
+
+  const handleKeyDown = (event: React.KeyboardEvent) => {
+    if ((event.key === 'Enter' || event.key === ' ') && checkType === 'radio') {
+      handleBoxClick();
+      event.preventDefault();
+    }
+  };
+
+  return (
+    <Items
+      onClick={handleBoxClick}
+      checked={checked}
+      $isFocused={focused}
+      tabIndex={0}
+      id={`item-${itemText}`}
```

Comment:
Using the text as `id` is not a good idea. Use a unique identifier instead

Reply:
done